### PR TITLE
rustdoc-json: Add some tests for typealias item

### PR DESCRIPTION
**Pull request by aDotInTheVoid**
r? @CraftSpider 

Improves https://github.com/rust-lang/rust/issues/81359

The test's arn't pretty, and I think eventually we need a better way of doing repeated tests on a deeply nested path,
without repeating the way to get to that path

@rustbot modify labels: +A-rustdoc-json +T-rustdoc +A-rustdoc +A-testsuite

### CODE REVIEW

File: src/tools/jsondocck/src/main.rs [hunk 1]
```diff
@@ -231,7 +231,21 @@ fn check_command(command: Command, cache: &mut Cache) -> Result<(), CkError> {
 
             let val = cache.get_value(&command.args[0])?;
             let results = select(&val, &command.args[1]).unwrap();
-            results.len() == expected
+            let eq = results.len() == expected;
+            if !command.negated && !eq {
+                return Err(CkError::FailedCheck(
+                    format!(
+                        "`{}` matched to `{:?}` with lenght {}, but expected lenght {}",
```
Comment (CraftSpider) [inline src/tools/jsondocck/src/main.rs:238]:
```suggestion
                        "`{}` matched to `{:?}` with length {}, but expected length {}",
```

**Comment by CraftSpider** [comment]
I was thinking about adding the ability to use `@set` on the left side of a check somehow recently. r=me with the nit fixed.

File: src/tools/jsondocck/src/main.rs (hunk 1 above)
Comment (aDotInTheVoid) [reply src/tools/jsondocck/src/main.rs]: Fixed and force pushed

**Comment by aDotInTheVoid** [comment]
> I was thinking about adding the ability to use @set on the left side of a check somehow recently.

Long term I'm not sure if the correct answer is do stuff like this to make jsonPath more sustainable, or to move to something like deserializing the json to a `Crate`, and then using normal rust tests on that.

Anyway, this is ready to be merged

**Comment by CraftSpider** [comment]
Hmm, that's an interesting idea I hadn't considered. The one downside is that it would probably require two files per test, but that may be worth it for just writing tests in Rust

**Comment by CraftSpider** [comment]
Anyways
@bors r+

**Pull request merged**